src/service/viewport-impl.js

```diff
@@ -1093,22 +1137,37 @@ export class ViewportBindingNaturalIosEmbed_ {
   }
 
   /** @override */
-  updatePaddingTop(paddingTop, opt_updateScrollPos, opt_lastPaddingTop) {
+  hideViewerHeader(transient, lastPaddingTop) {
+    if (!transient) {
+      this.updatePaddingTop(0);
+    } else {
+      // Add extra paddingTop to make the content stay at the same position
+      // when the hiding header operation is transient
+      onDocumentReady(this.win.document, doc => {
+        const existingPaddingTop =
+            this.win./*OK*/getComputedStyle(doc.body)['padding-top'] || '0';
+        doc.body.style.paddingTop =
```

Comment:
This is inside the `else` clause when the `hideViewerHeader` operation is transient. We add extra `paddingTop` only when the change is transient. We will remove the extra `paddingTop` by setting `body.style.paddingTop` to empty later when `showViewerHeader(false, '50px')` is called. The `false` means the `showViewerHeader` operation is not transient, and when the operation is not transient, we set `body.style.paddingTop` to empty.